### PR TITLE
Rename ExtendedOutput to BasicOutput

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by pvdrz — DISMISSED** [review]

**Review by Adam-Gleave — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

File: bee-message/src/output/foundry.rs [hunk 1]
```diff
@@ -227,7 +227,7 @@ impl FoundryOutput {
     ///
     #[inline(always)]
     pub fn address(&self) -> &Address {
-        // A FoundryOutput must have a AddressUnlockCondition.
+        // A FoundryOutput must have an AddressUnlockCondition.
```
Comment (grtlr) [inline bee-message/src/output/foundry.rs:230]: Maybe use doc comments here if we're touching it already.

File: bee-message/src/output/nft.rs [hunk 2]
```diff
@@ -168,7 +168,7 @@ impl NftOutput {
     ///
     #[inline(always)]
     pub fn address(&self) -> &Address {
-        // An NftOutput must have a AddressUnlockCondition.
+        // An NftOutput must have an AddressUnlockCondition.
```
Comment (grtlr) [inline bee-message/src/output/nft.rs:171]: Same

**Review by grtlr — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

File: bee-message/src/output/foundry.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-message/src/output/foundry.rs:230]: There are plenty of instances of missing link tbh, if you allow me I'll do it as part of the bee-message documentation PR

Comment (grtlr) [reply bee-message/src/output/foundry.rs:230]: Sure, that's why I already approved. :)

**Pull request merged**